src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs

```diff
@@ -592,9 +592,16 @@ IEnumerator IEnumerable.GetEnumerator()
             return new SeparatedSyntaxList<SyntaxNode>(nodes._list);
         }
 
+        [Obsolete("Implicit downcast is not safe. Use CastDown<TNode>() instead", error: true)]
         public static implicit operator SeparatedSyntaxList<TNode>(SeparatedSyntaxList<SyntaxNode> nodes)
         {
             return new SeparatedSyntaxList<TNode>(nodes._list);
         }
+
+        public SeparatedSyntaxList<TDerived> CastDown<TDerived>()
+            where TDerived : TNode
+        {
+            return new SeparatedSyntaxList<TDerived>(_list);
```

Comment:
> Since this type is immutable, we have the option of casting each element of _list here to force an InvalidCastException as early as possible

Right, we discarded that as an option due to not wanting a cast to be O(n) when n could be quite large.  

> Would renaming this method to UncheckedCastDown resolve your concern?

Possibly.  We'd need to take this through API review (for the new api), and compat council (for the obsoleting).  I wouldn't be opposed.  But i'm not strongly in favor either.

--

Also note: we'd need to do the same for SyntaxList as well, right?